src/hooks/useSearchQuery.jsx

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+export const useSearchQuery = () => {
```

Comment:
The first thing I don't like is the name because it is too generic. In fact, the hook is used only for making search query for cities search so let's name it properly.

`useCitySearchQuery` should be better